src/styles/_catalog.scss

```diff
@@ -0,0 +1,20 @@
+@use 'variables';
+
+.catalog {
+  padding: 50px 40px;
+  margin: 0;
+
+  display: grid;
+  justify-content: center;
+  gap: 46px 48px;
+
+  @each $width, $columns in variables.$screenWidth {
```

Comment:
Ensure that `variables.$screenWidth` is defined in the `variables` file and is structured as a map or list that can be iterated over with `@each`. If it's not defined or not structured correctly, this will cause an error during compilation.